aten/src/ATen/native/metal/mpscnn/MPSCNNOps.mm

```diff
@@ -510,6 +573,35 @@ Tensor upsample_nearest2d_vec(
   return output;
 }
 
+Tensor flatten_using_ints(
+    const Tensor& input,
+    int64_t start_dim,
+    int64_t end_dim) {
+  start_dim = maybe_wrap_dim(start_dim, input.dim());
+  end_dim = maybe_wrap_dim(end_dim, input.dim());
+  TORCH_CHECK(
+      start_dim <= end_dim,
+      "flatten() has invalid args: start_dim cannot come after end_dim");
+  std::vector<int64_t> shape;
+  if (input.dim() == 0) {
+    return input.reshape({1});
+  }
+  if (start_dim == end_dim) {
+    return input;
+  }
+  auto slice_numel =
+      prod_intlist(input.sizes().slice(start_dim, end_dim - start_dim + 1));
+  shape.reserve(input.dim() - end_dim + start_dim);
+  for (int64_t i = 0; i < start_dim; i++) {
+    shape.push_back(input.size(i));
+  }
+  shape.push_back(slice_numel);
+  for (int64_t i = end_dim + 1; i < input.dim(); i++) {
+    shape.push_back(input.size(i));
+  }
```

Comment:
Just cosmetic, maybe using std::copy instead of cycles:

```
auto isizes = input.sizes();
std::copy(isizes.begin(), isizes.begin() + start_dim, std::back_inserter(shape));
std::copy(isizes.begin() + end_dim + 1, isizes.end(), std::back_inserter(shape));
```